BASE/src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -676,6 +678,38 @@ public void Flush()
             }
         }
 
+        /// <summary>
+        /// Asynchronously Flushes the in-memory buffer and any metrics being pre-aggregated.
+        /// </summary>
+        /// <returns>
+        /// Returns true when telemetry data is transferred out of process (application insights server or local storage) and are emitted before the flush invocation.
+        /// Returns false when transfer of telemetry data to server has failed with non-retriable http status.
+        /// FlushAsync on InMemoryChannel always returns true, as the channel offers minimal reliability guarantees and doesn't retry sending telemetry after a failure.
+        /// </returns>
+        public Task<bool> FlushAsync(CancellationToken cancellationToken)
+        {
+            if (this.TryGetMetricManager(out MetricManager privateMetricManager))
+            {
+                privateMetricManager.Flush(flushDownstreamPipeline: false);
+            }
+
+            TelemetryConfiguration pipeline = this.configuration;
+            if (pipeline != null)
+            {
+                MetricManager sharedMetricManager = pipeline.GetMetricManager(createIfNotExists: false);
+                sharedMetricManager?.Flush(flushDownstreamPipeline: false);
+
+                ITelemetryChannel channel = pipeline.TelemetryChannel;
+
+                if (channel is IAsyncFlushable asyncFlushableChannel && !cancellationToken.IsCancellationRequested)
+                {
+                    return asyncFlushableChannel.FlushAsync(cancellationToken);
```

Comment:
Not really part of this PR, but we actually need to flush channels in *all* the sinks, as opposed to the default sink.
Please mark a TODO and we'll explore it after.